osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/SystemEventServiceTest.java

```diff
@@ -122,31 +183,77 @@ void verifyMaxValueReachedEvent() {
             messageMetadataCaptor.capture());
 
+    assertThat(invocationCounterIsLowered).isTrue();
+
     final RequestMessage requestMessage = requestMessageCaptor.getValue();
     assertThat(requestMessage.getDeviceIdentification())
-        .isEqualTo(device.getDeviceIdentification());
+        .isEqualTo(this.device.getDeviceIdentification());
     assertThat(requestMessage.getCorrelationUid()).isEqualTo("corr-id");
     assertThat(requestMessage.getOrganisationIdentification())
-        .isEqualTo(messageMetadata.getOrganisationIdentification());
-    assertThat(requestMessage.getIpAddress()).isEqualTo(messageMetadata.getIpAddress());
+        .isEqualTo(this.messageMetadata.getOrganisationIdentification());
+    assertThat(requestMessage.getIpAddress()).isEqualTo(this.messageMetadata.getIpAddress());
     assertThat(requestMessage.getRequest()).isInstanceOf(SystemEventDto.class);
 
     final SystemEventDto systemEventDto = (SystemEventDto) requestMessage.getRequest();
     assertThat(systemEventDto.getDeviceIdentification())
-        .isEqualTo(device.getDeviceIdentification());
+        .isEqualTo(this.device.getDeviceIdentification());
     assertThat(systemEventDto.getSystemEventType())
-        .isEqualTo(SystemEventTypeDto.INVOCATION_COUNTER_THRESHOLD_REACHED);
+        .isEqualTo(SystemEventTypeDto.ATTEMPT_TO_LOWER_INVOCATION_COUNTER);
     assertThat(systemEventDto.getTimestamp()).isNotNull();
 
     final MessageMetadata metadata = messageMetadataCaptor.getValue();
-    assertThat(metadata.getDeviceIdentification()).isEqualTo(device.getDeviceIdentification());
+    assertThat(metadata.getDeviceIdentification()).isEqualTo(this.device.getDeviceIdentification());
     assertThat(metadata.getCorrelationUid()).isEqualTo("corr-id");
     assertThat(metadata.getOrganisationIdentification())
-        .isEqualTo(messageMetadata.getOrganisationIdentification());
-    assertThat(metadata.getIpAddress()).isEqualTo(messageMetadata.getIpAddress());
+        .isEqualTo(this.messageMetadata.getOrganisationIdentification());
+    assertThat(metadata.getIpAddress()).isEqualTo(this.messageMetadata.getIpAddress());
     assertThat(metadata.getMessagePriority()).isEqualTo(MessagePriorityEnum.HIGH.getPriority());
     assertThat(metadata.getMessageType()).isEqualTo(MessageType.SYSTEM_EVENT.name());
-    assertThat(metadata.getDomain()).isEqualTo(messageMetadata.getDomain());
-    assertThat(metadata.getDomainVersion()).isEqualTo(messageMetadata.getDomainVersion());
+    assertThat(metadata.getDomain()).isEqualTo(this.messageMetadata.getDomain());
+    assertThat(metadata.getDomainVersion()).isEqualTo(this.messageMetadata.getDomainVersion());
+  }
+
+  @Test
+  void verifyHigherInvocationCounterReceived() throws IOException {
+    final DlmsConnection dlmsConnection = mock(DlmsConnection.class);
+    final DlmsConnectionManager connectionManager = mock(DlmsConnectionManager.class);
+    final long receivedHigherThanActualInvocationCounter = 100;
```

Comment:
The test gives the correct result, but there are some aspects that don't help me understand it and follow the logic.
One is the use of an invocation counter higher than the one that is used as the threshold related to another system event.
The other is that in tests where it is about the value of the invocation counter with the device entity versus the value of the invocation counter received from a physical device, it helps to see both of these values explicitly in the test method.